indra/sources/omnipath/processor.py

```diff
@@ -0,0 +1,175 @@
+            ptm_stmts.append(stmt)
+        print(Counter(unhandled_mod_types))
+        return ptm_stmts
+
+    def _stmt_from_op_lr(self, ligrec_json):
+        """Make ligand-receptor Complexes from Omnipath API interactions db"""
+        ligrec_stmts = []
+        ign_annot = {'source_sub_id', 'source', 'target', 'references'}
+        no_refs = 0
+        bad_pmid = 0
+        no_consensus = 0
+        if ligrec_json is None:
+            return ligrec_stmts
+
+        for lr_entry in ligrec_json:
+            if not lr_entry['references']:
+                no_refs += 1
+                continue
+            if len(lr_entry['sources']) == 1 and \
+                    lr_entry['sources'][0].lower() == 'protmapper':
+                logger.warning('Protmapper only source, skipping...')
+                continue
+
+            # Assemble evidence
+            evidence = []
+            for source_pmid in lr_entry['references']:
+                source_db, pmid = source_pmid.split(':')
+                if len(pmid) > 8:
+                    bad_pmid += 1
+                    continue
+                annot = {k: v for k, v in lr_entry.items() if k not in
+                         ign_annot}
+                annot['source_sub_id'] = source_db
+                evidence.append(Evidence(source_api='omnipath', pmid=pmid,
+                                         annotations=annot))
+
+            # Get complexes
+            ligrec_stmts.append(self._get_op_complex(lr_entry['source'],
+                                                     lr_entry['target'],
+                                                     evidence))
+
+            # On consensus, make Activations or Inhibitions as well
+            if bool(lr_entry['consensus_stimulation']) ^ \
+               bool(lr_entry['consensus_inhibition']):
+                activation = True if lr_entry['consensus_stimulation'] else \
+                    False
+                ligrec_stmts.append(self._get_ligrec_regs(
+                    lr_entry['source'], lr_entry['target'], evidence,
+                    activation=activation))
+            elif lr_entry['consensus_stimulation'] and \
+                    lr_entry['consensus_inhibition']:
+                no_consensus += 1
+
+        if no_refs:
+            logger.warning(f'{no_refs} entries without references were '
+                           f'skipped')
+        if bad_pmid:
+            logger.warning(f'{bad_pmid} references with bad pmids were '
+                           f'skipped')
+        if no_consensus:
+            logger.warning(f'{no_consensus} entries with conflicting '
+                           f'regulation were skipped')
+
+        return ligrec_stmts
+
+    @staticmethod
+    def _agent_from_up_id(up_id):
+        """Build an Agent object from a Uniprot ID. Adds db_refs for both
+        Uniprot and HGNC where available."""
+        db_refs = {'UP': up_id}
+        ag = Agent(None, db_refs=db_refs)
```

Comment:
We could set the `up_id` as an interim name here to avoid the corner case possibility of ending up with a None Agent name.